turbopack/crates/turbopack-core/src/module_graph/mod.rs

```diff
@@ -1451,10 +1451,17 @@ impl ModuleGraphRef {
             );
         }
 
+        let mut visit_order = 0usize;
+        let mut order = || {
+            let order = visit_order;
+            visit_order += 1;
+            order
```

Comment:
Normally I'm fine with shadowing in Rust, but the `order` variable name shadowing is maybe a little confusing here because the type is very different.

```suggestion
            let prev_order = visit_order;
            visit_order += 1;
            prev_order
```